core/sdk-core/src/main/java/software/amazon/awssdk/core/interceptor/ExecutionAttributes.java

```diff
@@ -48,11 +66,76 @@
         return this;
     }
 
+    /**
+     * Add all execution attributes from the provided collection to the internal collection of attributes.
+     */
+    public <U> ExecutionAttributes putAllAttributes(Map<ExecutionAttribute<?>, ?> executionAttributes) {
+        this.attributes.putAll(executionAttributes);
+        return this;
+    }
+
+    /**
+    * Merge attributes of a lower precedence into the current higher precedence current collection.
+    */
+    public <U> ExecutionAttributes merge(ExecutionAttributes lowerPrecedenceExecutionAttributes) {
```

Comment:
Please add a test for this

Reply:
this is tested by asyncClient_requestOverrideExecutionAttributesHavePrecedence test in ExecutionAttributesTest

Reply:
added an additional test for this which required some additional changes.

Reply:
I will add additional tests